app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserViewModel.kt

```diff
@@ -28,14 +27,30 @@ import com.infomaniak.lib.core.models.user.User
 import com.infomaniak.lib.login.ApiToken
 import com.infomaniak.mail.data.api.ApiRepository
 import com.infomaniak.mail.data.cache.mailboxInfos.MailboxController
+import com.infomaniak.mail.ui.main.user.SwitchUserAccountsAdapter.UiAccount
 import com.infomaniak.mail.utils.AccountUtils
+import com.infomaniak.mail.utils.sortMailboxes
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 import okhttp3.OkHttpClient
 
 class SwitchUserViewModel : ViewModel() {
 
-    fun fetchUsersMailboxes(users: List<User>) = viewModelScope.launch(Dispatchers.IO) {
+    fun listenToAccounts(): LiveData<List<UiAccount>> = liveData(Dispatchers.IO) {
+
+        val users = AccountUtils.getAllUsersSync()
+
+        updateMailboxes(users)
+
+        emitSource(MailboxController.getAllMailboxesAsync().map { mailboxes ->
+            users.map { user ->
+                UiAccount(user, mailboxes.list.filter { it.userId == user.id }.sortMailboxes())
```

Comment:
Use a realm query for this especially with realm you can also do the `sort` easily

Reply:
For the `filter`, I'm not sure if we can do it with Realm only.